src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertFromJsonCommand.cs

```diff
@@ -36,6 +36,13 @@ public class ConvertFromJsonCommand : Cmdlet
         [Parameter()]
         public SwitchParameter AsHashtable { get; set; }
 
+        /// <summary>
+        /// The maximum depth the json input is allowed to have. By default, there is no maximum.
+        /// </summary>
+        [Parameter]
+        [ValidateRange(ValidateRangeKind.Positive)]
+        public int? MaxDepth { get; set; }
```

Comment:
I think it should be:
```suggestion
        public int MaxDepth { get; set; } = int.MaxValue;
```

Reply:
Don't think we have *any* default cmdlets with nullable input types, so I don't see any need for that. I would think that since it's a `MaxDepth` parameter we could either opt for `uint` (because negative depth is insensible) *or* simply set the default to `-1` and have the code paths treat that as "no maximum depth".

Reply:
Underline type https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonSerializerSettings_MaxDepth.htm

Reply:
Yeah, you have a point there; a maximum depth of `0` is also pretty silly. However, given that the underlying json API uses `int` for these parameters, as @iSazonov pointed out, we can't safely use `uint` as it would allow the user to specify values that the underlying API can't handle. 

I think we need some clarification on whether the json API itself has a defined specification for "no maximum depth" and what that would look like.

@markekraus I know you worked on `Invoke-RestMethod` a good bit, do you have any insight to offer on the json API that might help here?

Reply:
@vexx32 I meant I could do `uint` with `[ValidateRange(1, int.MaxValue)]`, effectively not allowing user input values outside the underlying API.

Also, I was proposing `0` because the underlying API doesn't allow `0` for `MaxDepth`, only `1 - int.MaxValue` (it throws an exception at runtime, see [here](https://github.com/JamesNK/Newtonsoft.Json/blob/c356822a597ff2eb2a5b28e41c755ae040c1db10/Src/Newtonsoft.Json/JsonSerializerSettings.cs#L337)).

Reply:
Oh, that's just funny at that point. If you're going to throw *that* early... why not just make it `uint` haha!

Reply:
The underlying type is an int, so we should probably accept an int with the `[ValidateRange(1, int.MaxValue)]`. realistically, an JSON with a depth greater than `int32.MaxValue` is probably bad JSON to begin with.... A nullable type for this would be weird. accepting `-1` is also weird.